bftengine/src/bftengine/SeqNumInfo.cpp

```diff
@@ -405,6 +618,127 @@ IncomingMsgsStorage& SeqNumInfo::ExFuncForCommitCollector::incomingMsgsStorage(v
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+
+  return new FullCommitProofMsg(
+      r->getReplicasInfo().myId(), viewNumber, seqNumber, combinedSig, combinedSigLen, span_context);
+}
+
+InternalMessage SeqNumInfo::ExFuncForFastPathOptimisticCollector::createInterCombinedSigFailed(
+    SeqNum seqNumber, ViewNum viewNumber, const std::set<uint16_t>& replicasWithBadSigs) {
+  return FastPathCombinedCommitSigFailedInternalMsg(
+      seqNumber, viewNumber, CommitPath::OPTIMISTIC_FAST, replicasWithBadSigs);
+}
+
+InternalMessage SeqNumInfo::ExFuncForFastPathOptimisticCollector::createInterCombinedSigSucceeded(
+    SeqNum seqNumber,
+    ViewNum viewNumber,
+    const char* combinedSig,
+    uint16_t combinedSigLen,
+    const concordUtils::SpanContext& span_context) {
+  return FastPathCombinedCommitSigSucceededInternalMsg(
+      seqNumber, viewNumber, CommitPath::OPTIMISTIC_FAST, combinedSig, combinedSigLen, span_context);
+}
+
+InternalMessage SeqNumInfo::ExFuncForFastPathOptimisticCollector::createInterVerifyCombinedSigResult(SeqNum seqNumber,
+                                                                                                     ViewNum viewNumber,
+                                                                                                     bool isValid) {
+  return FastPathVerifyCombinedCommitSigResultInternalMsg(seqNumber, viewNumber, CommitPath::OPTIMISTIC_FAST, isValid);
+}
+
+uint16_t SeqNumInfo::ExFuncForFastPathOptimisticCollector::numberOfRequiredSignatures(void* context) {
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+  const ReplicasInfo& info = r->getReplicasInfo();
+  return 3 * info.fVal() + 2 * info.cVal() + 1;
+}
+
+std::shared_ptr<IThresholdVerifier> SeqNumInfo::ExFuncForFastPathOptimisticCollector::thresholdVerifier(
+    SeqNum seqNumber) {
+  return CryptoManager::instance().thresholdVerifierForOptimisticCommit(seqNumber);
+}
+
+concord::util::SimpleThreadPool& SeqNumInfo::ExFuncForFastPathOptimisticCollector::threadPool(void* context) {
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+  return r->getInternalThreadPool();
+}
+
+IncomingMsgsStorage& SeqNumInfo::ExFuncForFastPathOptimisticCollector::incomingMsgsStorage(void* context) {
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+  return r->getIncomingMsgsStorage();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// class SeqNumInfo::ExFuncForFastPathThresholdCollector
+///////////////////////////////////////////////////////////////////////////////
+
+FullCommitProofMsg* SeqNumInfo::ExFuncForFastPathThresholdCollector::createCombinedSignatureMsg(
+    void* context,
+    SeqNum seqNumber,
+    ViewNum viewNumber,
+    const char* const combinedSig,
+    uint16_t combinedSigLen,
+    const concordUtils::SpanContext& span_context) {
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+
+  return new FullCommitProofMsg(
```

Comment:
It would be good to check if this message is deleted by the caller of createCombinedSignatureMsg() function.

Reply:
This method is called internally by CollectorOfTrhesholdSignatures and the caller becomes the owner of the message.